packages/material-ui/test/integration/MenuList.test.js

```diff
@@ -258,266 +217,195 @@ describe('<MenuList> integration', () => {
+      fireEvent.keyDown(document.activeElement, { key: 'ArrowUp' });
+      expectSingleMenuItemFocusable(wrapper, 0);
+      expect(wrapper.getByRole('menu')).to.be.focused;
 
-      wrapper.simulate('keyDown', { key: 'ArrowUp' });
-      assertMenuItemTabIndexed(wrapper, 0);
-      assertMenuItemFocused(wrapper, 0);
+      fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+      expectSingleMenuItemFocusable(wrapper, 0);
+      expect(wrapper.getAllByRole('menuitem')[0]).to.be.focused;
     });
 
     it('should not wrap focus with ArrowDown from last', () => {
-      wrapper.simulate('keyDown', { key: 'ArrowDown' });
-      wrapper.simulate('keyDown', { key: 'ArrowDown' });
-      wrapper.simulate('keyDown', { key: 'ArrowDown' });
-      assertMenuItemTabIndexed(wrapper, 0);
-      assertMenuItemFocused(wrapper, 3);
-
-      wrapper.simulate('keyDown', { key: 'ArrowDown' });
-      assertMenuItemTabIndexed(wrapper, 0);
-      assertMenuItemFocused(wrapper, 3);
+      fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+      fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+      fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+      expectSingleMenuItemFocusable(wrapper, 0);
+      expect(wrapper.getAllByRole('menuitem')[3]).to.be.focused;
+
+      fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' });
+      expectSingleMenuItemFocusable(wrapper, 0);
+      expect(wrapper.getAllByRole('menuitem')[3]).to.be.focused;
     });
   });
 
   describe('MenuList with divider and disabled item', () => {
     let wrapper;
 
-    const resetWrapper = () => {
-      wrapper = mount(
+    before(() => {
+      wrapper = render(
         <MenuList>
           <MenuItem>Menu Item 1</MenuItem>
-          <Divider />
+          <Divider component="li" />
           <MenuItem>Menu Item 2</MenuItem>
           <MenuItem disabled>Menu Item 3</MenuItem>
           <MenuItem>Menu Item 4</MenuItem>
         </MenuList>,
       );
-    };
-
-    before(resetWrapper);
-
-    it('should skip divider and disabled menu item', () => {
-      wrapper.simulate('keyDown', { key: 'ArrowDown' });
-      assertMenuItemFocused(wrapper, 0);
-      wrapper.simulate('keyDown', { key: 'ArrowDown' });
-      assertMenuItemFocused(wrapper, 1);
-      wrapper.simulate('keyDown', { key: 'ArrowDown' });
-      assertMenuItemFocused(wrapper, 3);
-      wrapper.simulate('keyDown', { key: 'ArrowDown' });
-      assertMenuItemFocused(wrapper, 0);
-
-      wrapper.simulate('keyDown', { key: 'ArrowUp' });
-      assertMenuItemFocused(wrapper, 3);
-      wrapper.simulate('keyDown', { key: 'ArrowUp' });
-      assertMenuItemFocused(wrapper, 1);
-      wrapper.simulate('keyDown', { key: 'ArrowUp' });
-      assertMenuItemFocused(wrapper, 0);
-      wrapper.simulate('keyDown', { key: 'ArrowUp' });
-      assertMenuItemFocused(wrapper, 3);
     });
-  });
-
-  describe('MenuList with focusable divider', () => {
```

Comment:
While the implementation does allow this I don't think we should test those. After all this does seem invalid. See https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex#Accessibility_concerns

Reply:
My intent wasn't really to support focusable dividers, but rather to demonstrate focus management working for something that isn't a MenuItem but still meets the rules for recognizing it as a non-disabled, focusable child of the list. I still think there is value in the intent of this test, though it might be better to use something other than Divider for demonstrating it.

Reply:
I'm a bit cautious about opening this up to other components as well before we have a concrete use case. Especially since there is quite some movement in the react core with Flare and I would hope we'll be able to fully utilize their new event system without having to support other APIs.